prow/manifests/overlays/metal3/config.yaml

```diff
@@ -422,54 +422,6 @@ presubmits:
             image: docker.io/pipelinecomponents/markdownlint-cli2:0.9.0@sha256:71370df6c967bae548b0bfd0ae313ddf44bfad87da76f88180eff55c6264098c
             imagePullPolicy: Always
   metal3-io/baremetal-operator:
-    - name: gofmt
-      branches:
-        - main
-      skip_if_only_changed: '(((^|/)OWNERS)|(\.md))$'
-      decorate: true
-      spec:
-        containers:
-          - args:
-              - ./hack/gofmt.sh
-            command:
-              - sh
-            env:
-              - name: IS_CONTAINER
-                value: "TRUE"
-            image: docker.io/golang:1.21
-            imagePullPolicy: Always
-    - name: gofmt
-      branches:
-        - release-0.5
-        - release-0.4
```

Comment:
0.4 and 0.3 don’t have ”modern” golangci backport either? Is the old config really checking all? It was very broken and didn’t lint api or test at all etc?

Also, the new golangci is partial even in main, and not all improvements are backported to 0.5 AFAIK.

Reply:
Oh ok, so we may still need this on the release branches? Maybe I should just remove jobs for main. The rest can live until the release branch is dropped from testing :slightly_smiling_face:

Reply:
0.5 looks like it has enalbed gosec/govet/gofmt without all the backports and has the action, so it can go. 0.4 and 0.3 need to stay IMO. They'll be out of support in 2 months anyways.

Reply:
@tuminoid for some reason I dont see the workflow file in 0.5 branch either, are you sure the golangci-lint  action is active on that branch ? IMO we should only remove from main branch.

Reply:
It is not an action on release-0.5. It is [golint](https://github.com/metal3-io/project-infra/blob/7b988cb551aef97114c0864b33bdd8c166b0ec06/prow/manifests/overlays/metal3/config.yaml#L685-L702), which calls [golint.sh](https://github.com/metal3-io/baremetal-operator/blob/release-0.5/hack/golint.sh), which calls [make lint](https://github.com/metal3-io/baremetal-operator/blob/72057304eb6fa55cbe3f5b00c5665748fe442dfe/Makefile#L126-L129). This is using golangci-lint and has [gofmt enabled](https://github.com/metal3-io/baremetal-operator/blob/72057304eb6fa55cbe3f5b00c5665748fe442dfe/.golangci.yml#L8-L11)